test/controllers/offer.controller.test.js

```diff
@@ -1,677 +1,815 @@
-//       request()
-//         .get(`/api/v1/offer/${offerId}`)
-//         .set('authorization', userToken)
-//         .end((err, res) => {
-//           expect(res).to.have.status(404);
-//           expect(res.body.success).to.be.eql(false);
-//           expect(res.body.message).to.be.eql('Invalid token');
-//           done();
-//         });
-//     });
-//   });
-
-//   context('with an invalid offer id', () => {
-//     const invalidId = mongoose.Types.ObjectId();
-//     it('returns not found', (done) => {
-//       request()
-//         .get(`/api/v1/offer/${invalidId}`)
-//         .set('authorization', userToken)
-//         .end((err, res) => {
-//           expect(res).to.have.status(404);
-//           expect(res.body.success).to.be.eql(false);
-//           done();
-//         });
-//     });
-//   });
-
-//   context('without token', () => {
-//     it('returns error', (done) => {
-//       request()
-//         .get(`/api/v1/offer/${offerId}`)
-//         .end((err, res) => {
-//           expect(res).to.have.status(403);
-//           expect(res.body.success).to.be.eql(false);
-//           expect(res.body.message).to.be.eql('Token needed to access resources');
-//           done();
-//         });
-//     });
-//   });
-
-//   context('when getOffer service fails', () => {
-//     it('returns the error', (done) => {
-//       sinon.stub(Offer, 'aggregate').throws(new Error('Type Error'));
-//       request()
-//         .get(`/api/v1/offer/${offerId}`)
-//         .set('authorization', userToken)
-//         .end((err, res) => {
-//           expect(res).to.have.status(500);
-//           done();
-//           Offer.aggregate.restore();
-//         });
-//     });
-//   });
-// });
+import mongoose from 'mongoose';
+import { expect, request, sinon, useDatabase } from '../config';
+import Offer from '../../server/models/offer.model';
+import User from '../../server/models/user.model';
+import OfferFactory from '../factories/offer.factory';
+import EnquiryFactory from '../factories/enquiry.factory';
+import UserFactory from '../factories/user.factory';
+import PropertyFactory from '../factories/property.factory';
+import { createOffer } from '../../server/services/offer.service';
+import { addEnquiry } from '../../server/services/enquiry.service';
+import { addUser } from '../../server/services/user.service';
+import { addProperty } from '../../server/services/property.service';
+import { OFFER_STATUS } from '../../server/helpers/constants';
+
+useDatabase();
+
+let adminToken;
+let userToken;
+const userId = mongoose.Types.ObjectId();
+const adminId = mongoose.Types.ObjectId();
+const adminUser = UserFactory.build({ _id: adminId, role: 0, activated: true });
+const regualarUser = UserFactory.build({ _id: userId, role: 1, activated: true });
+const propertyId = mongoose.Types.ObjectId();
+const property = PropertyFactory.build({ _id: propertyId, addedBy: adminId, updatedBy: adminId });
+
+describe('Offer Routes', () => {
```

Comment:
The test is for Controllers not Routes
```suggestion
describe('Offer Controller', () => {
```